#!/usr/bin/env python3
"""
Teste R√°pido de Conex√£o gRPC
"""

import sys
import os
import grpc

# Configura√ß√£o de ambiente
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

# Imports gRPC
from grpc_services import services_pb2, services_pb2_grpc

def test_cursos_connection():
    """Testa conex√£o com servi√ßo de cursos"""
    try:
        # Conecta com o servi√ßo
        channel = grpc.insecure_channel('localhost:8081')
        stub = services_pb2_grpc.CursosServiceStub(channel)
        
        # Testa status
        response = stub.GetCursosStatus(services_pb2.Empty(), timeout=5)
        print(f"‚úÖ CURSOS: {response.message}")
        
        # Testa funcionalidade
        request = services_pb2.GetUserCoursesRequest(id_usuario="test_123")
        response = stub.GetUserCourses(request, timeout=5)
        print(f"‚úÖ FUNCIONALIDADE: Encontrados {len(response.cursos_inscritos)} cursos")
        
        channel.close()
        return True
        
    except Exception as e:
        print(f"‚ùå ERRO: {e}")
        return False

if __name__ == '__main__':
    print("üß™ TESTE R√ÅPIDO DE CONEX√ÉO gRPC")
    print("=" * 40)
    
    if test_cursos_connection():
        print("\nüéâ SERVI√áO DE CURSOS FUNCIONANDO!")
    else:
        print("\n‚ùå Falha na conex√£o")